locale/en/docs/guides/diagnostics-flamegraph.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Diagnostics - Flame Graphs
+layout: docs.hbs
+---
+
+# Flame Graphs
+
+## What's a flame graph useful for?
+
+Flame graphs are a way of visualizing CPU time spent in functions. They can help you pin down where you spend too much time doing synchronous operations.
+
+## How to create a flame graph 
+
+You might have heard creating a flame graph for Node.js is difficult, but that's not true (anymore).
+Solaris vms are no longer needed for flame graphs!
+
+Flame graphs are generated from `perf` output, which is not a node-specific tool. While it's the most powerful way to visualize CPU time spent, it may have issues with how JavaScript code is optimized in Node 8 and above. See [perf output issues](#perf-output-issues) section below.
+
+### Use a pre-packaged tool
+
+If you want a single step that produces a flame graph locally, try [0x](https://www.npmjs.com/package/0x)
+
+For diagnosing production deployments, read these notes: [0x production servers](https://github.com/davidmarkclements/0x/blob/master/docs/production-servers.md)
+
+### Create a flame graph with system perf tools
+
+The purpose of this guide is to show steps involved in creating a flame graph and keep you in control of each step.
+
+If you want to understand each step better take a look at the sections that follow were we go into more detail.
+
+Now let's get to work.
+
+1. Install `perf` (usually available through the linux-tools-common package if not already installed)
+2. try running `perf` - it might complain about missing kernel modules, install them too
+3. run node with perf enabled (see [perf output issues](#perf-output-issues) for tips specific to node versions)
+```bash
+perf record -e cycles:u -g -- node --perf-basic-prof app.js
+``` 
+4. disregard warnings unless they're saying you can't run perf due to missing packages; you may get some warnings about not being able to access kernel module samples which you're not after anyway.
+5. Run `perf script > perfs.out` to generate the data file you'll visualize in a moment. It's useful to [apply some cleanup](#filtering-out-node-internal-functions) for a more readable graph 
+6. install stackvis if not yet installed `npm i -g stackvis`
+7. run `stackvis perf < perfs.out > flamegraph.htm`
```

Comment:
I tried this today and the result does not display nicely for me in the browser.  Does it still work ok for you?

Reply:
Maybe we should suggest using [Brendan Gregg's flamegraph scripts](http://www.brendangregg.com/blog/2014-09-17/node-flame-graphs-on-linux.html) instead? Stackvis result is quite unreadable here as well:

`stackvis perf < perfs.out > flamegraph.htm `:

![image](https://user-images.githubusercontent.com/4048656/46082110-a85f8700-c174-11e8-8e12-8c6aa0e3a91c.png)

`./FlameGraph/stackcollapse-perf.pl  < ./perfs.out  | ./FlameGraph/flamegraph.pl --color=js > ./flamegraph.svg`:

![image](https://user-images.githubusercontent.com/4048656/46082257-0e4c0e80-c175-11e8-9e23-13c4e01432de.png)

Reply:
I did follow that suggestion in the demo I was putting together for my NodeConfEU presentation.  FlameGraph worked for me.  Once we close on the content we should see if we can figure out some testing to make sure our recommendation continues to work properly.

Reply:
That's exactly why I'm not saying this is ready to publish - it started showing me the same issue you have on the screenshot. It seemed an issue with perf output, so I upgraded my OS and retired, got the same thing, ran out of time. I'll check other options, but it's good to see the output from perf seems ok and it's about the visualization tool.

Reply:
@naugtur thanks for the update.

Reply:
I've compared various flamegraph generators and the difference in readabiliy is mostly due to overflow:hidden of long trace names, but the input data is still incorrect. I switched to ubuntu 18.04.1 and it still produces bad stack traces. 

Planning to check on a few systems and compare.

Reply:
> but the input data is still incorrect

Incorrect how? If you could share an example I can happily take a look :)